### PR TITLE
osd/SnapManager: Crimson - fix std::string constructed from 0

### PR DESCRIPTION
**Pull request by zhscn**
the output of clang 17:
```
/home/zhangsong02/nv/ceph/src/osd/SnapMapper.cc:150:59: error: attempt to use a deleted function
  150 |     *next_or_current = std::make_pair(key, std::move(vals[0]));
      |                                                           ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/basic_string.h:742:7: note: 'basic_string' has been explicitly marked deleted here
  742 |       basic_string(nullptr_t) = delete;
      |       ^
1 error generated.
```

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by Matan-B — APPROVED** [review]

File: src/osd/SnapMapper.cc [hunk 1]
```diff
@@ -147,7 +147,7 @@ int OSDriver::get_next_or_current(
     ch, hoid, FuturizedStore::Shard::omap_keys_t{key}
   ).safe_then([&key, next_or_current] (FuturizedStore::Shard::omap_values_t&& vals) {
     assert(vals.size() == 1);
-    *next_or_current = std::make_pair(key, std::move(vals[0]));
```
Comment (rzarzynski) [inline src/osd/SnapMapper.cc:150]: `omap_values_t` is:

```cpp
    using omap_values_t = std::map<std::string, ceph::bufferlist, std::less<>>;
```

so we definitely want the `begin()->second`.
Good catch, thanks!

**Review by rzarzynski — APPROVED** [review]

**Review by liu-chunmei — APPROVED** [review]

**Review by athanatos — APPROVED** [review]

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
```
7 - run-tox-mgr-dashboard-py3 (Failed)
```

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2024-03-05_16:18:08-crimson-rados-wip-matanb-crimson-testing-march-5-distro-crimson-smithi/

`7582495` - osd.1 - https://tracker.ceph.com/issues/64009
`7582487` - osd.0 - https://tracker.ceph.com/issues/64009

**Pull request merged**